rust/ql/lib/codeql/files/FileSystem.qll

```diff
@@ -47,6 +48,7 @@ class File extends Container, Impl::File {
         exists(AstNode node, Location loc |
           not node instanceof Comment and
           not node instanceof SourceFile and
+          not getImmediateParent(node) instanceof SourceFile and // ignore top-level elements for now as we're getting their locations wrong when a comment is attached
```

Comment:
Perhaps just `not node instanceof Item` . That should also deal with `Item`s that are nested in functions.

Reply:
Done.

Though I am a bit worried the attached comments issue is going to turn out to affect things other than top-level items, I'll be happier when the issue with locations itself is fixed.